core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -98,35 +151,73 @@ public GeoBoundingBoxQueryBuilder bottomRight(String geohash) {
      * @param lon The longitude
      */
     public GeoBoundingBoxQueryBuilder bottomLeft(double lat, double lon) {
-        box[BOTTOM] = lat;
-        box[LEFT] = lon;
+        if (topLeft != null) {
+            topLeft.resetLon(lon);
+        } else {
+            topLeft = new GeoPoint(Double.NaN, lon);
+        }
+
+        if (bottomRight != null) {
+            bottomRight.resetLat(lat);
+        } else {
+            bottomRight = new GeoPoint(lat, Double.NaN);
+        }
```

Comment:
I get sooo confused when I read this method. Is it just because of my lack of geo knowledge? :) What are we doing here exactly?